framer/Material-UI.framerfx/code/Avatar.tsx

```diff
@@ -29,7 +27,7 @@ const defaultProps: Props = {
   height: 40,
 };
 
-export const Avatar: React.SFC<Props> = (props: Props) => {
+export function Avatar(props: Props): JSX.Element {
```

Comment:
Not sure why you consider it "interesting" when this is the exact way we write function components in the rest of the monorepo. Maybe I missed the issue where you want to enforce this throughout the repo but as far as I can tell this is unrelated to this PR. 

I was simply bringing it in line with the rest of the repo while fixing the issues with React.FunctionComponent.

Reply:
By "interesting", I meant as bringing a viable option. We have a standard around how we want to write components in React (we enforce it here) but do we have a standard yet around how we type them?

From what I understand material-ui-x and material-ui-pickers have more components written in TypeScript than the main repository, hence this could be a good opportunity to agree on a standard for typing them (I don't have my point of view on which one is should be). I think that it should also convert the React.forwardRef usage.

For instance, do we need to use `React.FC` in?
https://github.com/mui-org/material-ui-pickers/blob/4d762e7868c480ce6067645512a425fa7a1f6ee4/lib/src/views/Calendar/Day.tsx#L136
https://github.com/mui-org/material-ui-x/blob/da5e83aa69291821e14218184d05a2c1c313bfd6/packages/grid/x-grid-data-generator/src/renderer/done-renderer.tsx#L6

Reply:
(sorry for hijacking the pull request, happy to move it to an RFC)

Reply:
Implicit can be problematic because of TypeScript. In React you can return a string but TypeScript won't consider it a component. Explicit return type is something I always do because it improves readability. For components it's somewhat trivial but doing it everywhere is just muscle memory. I don't think we should enforce any rule around it for the sake of having a rules. Rules are only useful if they solve a problem. So you need to show me first that unifying any two signatures solves a given problem.

Reply:
Why `React.FC` might be problematic:
- has some issues with defaultProps: https://github.com/typescript-cheatsheets/react-typescript-cheatsheet/issues/87
- adds implicit `children`

Reply:
Marking as resolve so it doesn't clutter the PR. We can always discuss this somewhere else. There's unfortunately a lot of obscure issue with either type approach.